### DISCUSSION TITLE
Where should I add spec for insert_many for this case?

### DISCUSSION THREAD

**Discussion by midnight-wonderer** [mongodb/mongo-ruby-driver#2430] [Q&A] [answered]
Hello,
I use undocumented features, and I wanted to ensure it won't break in the future.
I am here to help add test cases.
The feature is `insert_many` for enumerable.

~~~ruby
collection.insert_many((1..100).lazy.map{ |s| { _id: "id_#{s}" } } )
~~~

`(1..100).lazy.map{ |s| {_id: "id_#{s}"}}` is not an array but enumerable. The driver seems to work anyway because it did not use any of the array methods.
Where should I add test cases related to this one?

**Comment by p-mongo** [comment]
Hi @midnight-wonderer , thanks for considering this. The test can go into https://github.com/mongodb/mongo-ruby-driver/blob/master/spec/mongo/collection_crud_spec.rb#L351.

I also filed https://jira.mongodb.org/browse/RUBY-3087 to track this request.

**Comment by alexbevi** [comment, marked as answer]
[RUBY-3087](https://jira.mongodb.org/browse/RUBY-3087) was delivered in 2.18.2 of the driver :)
